blues/ncmc.py

```diff
@@ -181,59 +189,45 @@ def __init__(self, temperature, residueList, **kwds):
         beta = 1.0 / kT
         self.beta = beta
 
-    def get_particle_masses(self, system, residueList=None):
-        if residueList == None:
-            residueList = self.residueList
-        mass_list = []
-        total_mass = 0*unit.dalton
-        for index in residueList:
-            mass = system.getParticleMass(int(index))
-            total_mass = total_mass + mass
-#            print('mass', mass, 'total_mass', total_mass)
-            mass_list.append([mass])
-        total_mass = np.sum(mass_list)
-        mass_list = np.asarray(mass_list)
-        mass_list.reshape((-1,1))
-        total_mass = np.array(total_mass)
-        total_mass = np.sum(mass_list)
-        temp_list = np.zeros((len(residueList), 1))
-        for index in range(len(residueList)):
-            mass_list[index] = (np.sum(mass_list[index])).value_in_unit(unit.daltons)
-        mass_list =  mass_list*unit.daltons
-        self.total_mass = total_mass
-        self.mass_list = mass_list
-        return total_mass, mass_list
-            
+    def get_particle_masses(self, system, atomsIdx):
+        masses = unit.Quantity(np.zeros([len(atomsIdx),1],np.float32), unit.dalton)
+        #system = context.getSystem()
+        for ele, idx in enumerate(atomsIdx):
+            masses[ele] = system.getParticleMass(idx)
+        self.total_mass = masses.sum()
+        self.mass_list = masses
+        return self.total_mass, self.mass_list
+
     def zero_masses(self, system, atomList=None):
         """
         Zeroes the masses of specified atoms to constrain certain degrees of freedom
         Arguments
         ---------
-        system: simtk.openmm.system 
+        system: simtk.openmm.system
             system to zero masses
-        atomList: list of ints 
+        atomList: list of ints
             atom indicies to zero masses
         """
         for index in (atomList):
             system.setParticleMass(index, 0*unit.daltons)
-        
+
 
     def calculate_com(self,  pos_state, total_mass=None, mass_list=None, residueList=None, rotate=False):
         """
-        This function calculates the com of specified residues and optionally 
+        This function calculates the com of specified residues and optionally
```

Comment:
@sgill2 (and @nathanmlim) - can we not use "com" as short for "center of mass" without defining that the first time it's used in the doc string?